### PR TITLE
Add atlas interface

### PR DESCRIPTION
**Pull request by lukasm91**
- defines the current requirements of the interface (currently in `src/dawn/CodeGen/CXXNaive-ico/CXXNaiveCodeGen.cpp`)
- Adapts the `mylib` interface to this interface
- Implements the interface for `atlas` and adds a driver code
- Generalize the unstructured "CodeGenPlayground" to  take into account that different nodes might have different number of neighbours
- Fix initialization in the periodic unstructured example
- (unrelated) add bundle-folder to ignorelist of `clang-format-all.sh`

### CODE REVIEW

**Comment by lukasm91** [comment]
launch jenkins

File: dawn/prototype/README [hunk 1]
```diff
@@ -3,5 +3,7 @@ Compiling and running the prototype (until we have a proper setup)
 1. Compile Dawn
 2. cd <path/to/dawn> && ./bundle/build/dawn-prefix/src/dawn-build/bin/unittest/DawnCUnittest --gtest_also_run_disabled_tests 2>./prototype/generated.hpp && clang-format -i ./prototype/generated.hpp
    --> This will run the disabled CodeGenPlayground test that generated and outputs code
-3. cd prototype && g++ -o out driver.cpp grid.cpp -o out -std=c++17 -I <path/to/gtclang>/src; ./out
-4. paraview of_..vtk
+3. For "lukas_lib": cd prototype && g++ -o out driver.cpp grid.cpp -o out -std=c++17 -I<path/to/gtclang>/src; ./out
```
[the remark this answers is not in the document]
Comment (havogt) [reply dawn/prototype/README]: Also `my_interface.hpp` doesn't exist. How is it supposed to work?

File: dawn/prototype/atlas_interface.hpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+} // namespace utility
+
+namespace atlasInterface {
+
+struct atlasTag {};
+
+template <typename T>
+class Field {
+public:
+  T operator()(int f) const { return atlas_field_(f, 0); }
```
Comment (havogt) [inline dawn/prototype/atlas_interface.hpp]: Why not return `const&`

File: dawn/src/dawn/CodeGen/CodeGen.cpp [hunk 3]
```diff
@@ -29,10 +29,12 @@ std::string CodeGen::generateGlobals(stencilInstantiationContext& context,
                                      std::string outer_namespace_, std::string inner_namespace_) {
 
   std::stringstream ss;
-  Namespace outerNamespace(outer_namespace_, ss);
   std::string globals = generateGlobals(context, inner_namespace_);
-  ss << globals;
-  outerNamespace.commit();
+  if(globals != "") {
```
Comment (havogt) [inline dawn/src/dawn/CodeGen/CodeGen.cpp]: I propose std::optional instead of an empty string

Comment (lukasm91) [reply dawn/src/dawn/CodeGen/CodeGen.cpp]: `globals` is generated code and we want to wrap it into a namespace. The namespace is of course not needed if there is no code - that would be just an empty namespace. So the code expresses quite well what we want to do.

**Comment by lukasm91** [comment]
launch jenkins

**Pull request merged**